### DISCUSSION TITLE
Getting a parse error with blank message

### DISCUSSION THREAD

**Discussion by bfg-coding** [helix-editor/helix#5113] [Q&A]
Yuck is a configuration language for the EWW to create desktop widgets. I'm also the one working on the tree-sitter for Yuck (https://github.com/bfg-coding/tree-sitter-yuck).

I really need assistance on getting Helix parser working as I'm getting an error that isn't giving me any direction on where the issue is.  Here is my fork that has everything setup (https://github.com/helix-editor/helix/compare/master...bfg-coding:helix:master)  assistance and understanding would be very much appreciated.

```
 This query could not be parsed: QueryError { row: 29, column: 3, offset: 497, message: "", kind: NodeType }
```

**Comment by the-mikedavis** [comment]
The `!=` node on line 30 of the `highlights.scm` is not a valid node exposed by the grammar. As a rule of thumb, any string literal in the grammar.js file (for example the `'?:'` on [this line](https://github.com/bfg-coding/tree-sitter-yuck/blob/5f521fef89ce7d06b00330b680d8a3183829df2c/grammar.js#L108)) can be written as a string literal in the queries. If I comment out line 30, I see the queries working. Unfortunately we can't improve this error message - it's emitted directly by the tree-sitter library.

As a side-note, the `source.git` url is not valid [here](https://github.com/helix-editor/helix/blob/d9f806e0b9cdc6e880b792e1dc7619edbd261ab8/languages.toml#L1967) in the languages.toml: you can use `source.path = "/path/to/tree-sitter/grammar/repository"` instead to use a path rather than a remote git repository.

**Reply by bfg-coding** [reply]
This is exactly what I needed.  The big issue is I didn't know exactly which file it was was referencing in the log file. I understand that the errors may be coming directly from tree-sitter so it's hard to decipher or add context.

**Comment by Philipp-M** [comment]
Hey, FYI I've just created a tree-sitter parser as well for yuck, see  #6064
The parser closely follows the original lalrpop parser of eww, I had to add an external scanner for string literals (for lookahead parsing string interpolation and escaped characters).

But I think that maybe syntax highlighting itself *may* be improved, e.g. extra highlighting of all the different widget symbols, it currently is rather basic (but I think at least every bit of code should be highlighted).
